src/templates/post.html

```diff
@@ -23,52 +27,91 @@
                 <a href="javascript:quote('{{post.no}}');" title="Reply to this post">{{post.no}}</a>
             </span>
         </div>
+		{% if post.resto == 0 and post.filename %}
+        <div class="file" id="f{{post.no}}">
+            <div class="fileText" id="fT{{post.no}}">
+                File:
+                <a href="{{ thumb_uri.format(board_shortname=post.board_shortname) }}/{{ (post.asagi_preview_filename|string)[0:4] }}/{{ (post.asagi_preview_filename|string)[4:6] }}/{{ post.asagi_preview_filename}}">{{post.filename}}.{{post.ext}}</a>
+                (<span title="{{post.md5}}">{% if post.spoiler %}Spoiler, {% endif %}{% if post.fsize >= 1048576 %}{{(post.fsize / 1024 / 1024 )| round(2, 'floor')}} MB, {{post.w}}x{{post.h}}){% else %}{{(post.fsize / 1024 )| round(1, 'floor')}} KB, {{post.w}}x{{post.h}}){% endif %}</span>
+				{% if search_result %}
+					Replies: {{post.replies}} Files: {{post.images}} {% if post.posters %}Posters: {{post.posters}}{% endif %}
+				{% endif %}
+            </div>
+            <a class="fileThumb" href="{{ image_uri.format(board_shortname=post.board_shortname) }}/{{ (post.asagi_filename|string)[0:4] }}/{{ (post.asagi_filename|string)[4:6] }}/{{ post.asagi_filename}}" target="_blank">
+                <img src="{{ thumb_uri.format(board_shortname=post.board_shortname) }}/{{ (post.asagi_preview_filename|string)[0:4] }}/{{ (post.asagi_preview_filename|string)[4:6] }}/{{ post.asagi_preview_filename}}" data-md5="{{post.md5}}" style="height: {{post.tn_h}}px; width: {{post.tn_w}}px;" class="opImg postImg{% if post.spoiler %} spoiler{% endif %}" loading="lazy">
+            </a>
+        </div>
+        {% endif %}
         <div class="postInfo desktop" id="pi{{post.no}}">
-            {% if search_result %}
+			{% if search_result %}
                 <b>/{{post.board_shortname}}/</b>
             {% endif %}
             <input type="checkbox" name="{{post.no}}" value="delete">
-            <span class="nameBlock">
-                <span class="name">{{post.name}}</span>
-            </span>
+				{% if post.filedeleted and not post.deltime %}
+					<strong class="warning" title="This post was prematurely deleted.">[Deleted]</strong>
+				{% endif %}
+				{% if post.filedeleted and post.deltime %}
+					<strong class="warning" title="This post was deleted on {{post.deltime}}.">[Deleted]</strong>
+				{% endif %}
+			{% if post.sub %}
+					<span class="subject">{{post.sub}}</span>
+			{% endif %}
+            <span class="nameBlock{% if post.capcode == 'A' %} capcodeAdmin{% endif %}{% if post.capcode == 'F' %} capcodeFounder{% endif %}{% if post.capcode == 'M' %} capcodeMod{% endif %}{% if post.capcode == 'D' %} capcodeDeveloper{% endif %}{% if post.capcode == 'G' %} capcodeManager{% endif %}{% if post.capcode == 'V' %} capcodeVerified{% endif %}">
```

Comment:
I have the same comment as with the `thread.capcode` logic.  You can make one macro that is used for both of these. See how I used `{% macro table(list_of_dicts) %}` in `macros.html`.